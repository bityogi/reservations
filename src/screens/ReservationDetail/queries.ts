import { reservationQuery } from '../../util/queries';
import { client } from '../../util/requests';

export const getReservation = async (id) => {
  
  const { data: { reservation }} = await client.query({ 
    query: reservationQuery,
    variables: {
      where: {
        id
      }
    }
  });

  return {
    ...reservation,
    arrivalDate: new Date(reservation.arrivalDate),
    departureDate: new Date(reservation.departureDate)
  }
} 