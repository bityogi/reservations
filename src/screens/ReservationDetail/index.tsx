import React, { PureComponent } from 'react';
import {
  View,
  Text,
  StyleSheet
} from 'react-native';
import { HeaderBackButton } from 'react-navigation';
import { isEmpty } from 'lodash';

import { getReservation } from './queries';
import { formatDate } from '../../util';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  data: {
    marginTop: 20,
    marginBottom: 20,
  },
  name: {
    fontSize: 30,
  },
  hotelName: {
    fontSize: 20,
    fontWeight: '500',
    color: '#bdbdbd'
  },
  date: {
    fontSize: 15,
    fontWeight: '300'
  }
})

class ReservationDetail extends PureComponent {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'Reservation Detail',
      headerLeft:(<HeaderBackButton onPress={() => navigation.navigate('Reservations', { refresh: true })}/>)
    }
  };

  state = {
    reservation: {}
  }
  

  componentDidMount() {
    const { navigation } = this.props;
    const reservationId = navigation.getParam('reservationId')

    if (reservationId) {
      getReservation(reservationId)
        .then(reservation => {
          this.setState({ reservation })
        })
    }
  }

  render() {
    const { reservation } = this.state;

    if (isEmpty(reservation)) {
      return null
    }
    return (
      <View style={styles.container}>
        <Text style={[styles.data, styles.name]}>{reservation.name}</Text>
        <Text style={[styles.data, styles.hotelName]}>{reservation.hotelName}</Text>
        <Text style={[styles.data, styles.date]}>{formatDate(reservation.arrivalDate)}</Text>
        <Text style={[styles.data, styles.date]}>{formatDate(reservation.departureDate)}</Text>
      </View>
    )
  }
}

export default ReservationDetail;