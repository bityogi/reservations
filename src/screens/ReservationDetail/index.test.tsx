import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import { shallow } from 'enzyme';
import ReservationDetail from './index';

jest.mock("NativeModules", () => ({
  UIManager: {
    RCTView: () => ({
      directEventTypes: {}
    })
  },
  KeyboardObserver: {},
  RNGestureHandlerModule: {
    attachGestureHandler: jest.fn(),
    createGestureHandler: jest.fn(),
    dropGestureHandler: jest.fn(),
    updateGestureHandler: jest.fn(),
    State: {},
    Directions: {}
  }
}));


jest.mock('../../util/requests');
jest.mock('./queries');

const navigationProp = {
  navigate: (to: string, options?: object) => { console.log('Navigating') },
  getParam: (param: string) => { return '1' },
  addListener: (listenEvent: string, payload: object) => {}
}

describe('ReservationDetail', () => {

  it('should render correctly', () => {
    const component = shallow(<ReservationDetail navigation={navigationProp} />);

    expect(component).toMatchSnapshot();
  })
})