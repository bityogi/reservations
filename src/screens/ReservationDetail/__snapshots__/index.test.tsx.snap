// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationDetail should render correctly 1`] = `""`;
